hack/download-charts.sh

```diff
@@ -38,12 +49,24 @@ function patchIstioCharts() {
 - apiGroups: ["security.openshift.io"] \
   resources: ["securitycontextconstraints"] \
   resourceNames: ["privileged"] \
-  verbs: ["use"]/' "${HELM_DIR}/istio-cni/templates/clusterrole.yaml"
+  verbs: ["use"]/' "${CHARTS_DIR}/istio-cni/templates/clusterrole.yaml"
+}
+
+function convertIstioProfiles() {
```

Comment:
Do we really need to do that? I'm worried this operator becomes 2.x with lots of download/patch/sed/hack boilerplate code.

Can't we just define our own set of profiles, based on upstream?

Reply:
We do. We need to convert from `IstioOperator.install.istio.io` to `Istio.operator.istio.io` and remove the fields that we don't support (yet?), such as `components`, `meshConfig`, `hub`, and `tag`.

That's exactly what we're doing. We're defining our own set based on upstream.

Reply:
Sorry, I meant, should we just have static files in our repo instead of patching istio/istio files?

Reply:
Why? Then we'd have to update them manually whenever they change upstream. IMHO we should support upstream profiles, not create our own.

Reply:
We are already getting only the spec.values map within `getValues()`. Can't we just use the raw upstream file as-is and ignore everything that's not `spec.values` in it?

Reply:
Using the upstream file would be misleading and confusing because:
- the upstream file is an IstioOperator manifest
- ignoring fields will make things confusing and misleading; users will waste time figuring out why something isn't working

Also, this probably isn't the final state of these files. But we can't yet know what the final state will be. The way these files are transformed in this PR is what makes sense currently. We will most likely change them eventually.